lib/convection/model/template.rb

```
@@ -183,8 +183,75 @@ def diff(other, stack_ = nil)
       end
 
       def to_json(stack_ = nil, pretty = false)
-        return JSON.generate(render(stack_)) unless pretty
-        JSON.pretty_generate(render(stack_))
+        stack = render(stack_)
+
+        #bytesize
+        json = JSON.generate(stack)
+        template_bytesize = json.bytesize
+        cf_max_bytesize = 51200
+        if template_bytesize > cf_max_bytesize
+          Kernel.abort("Error: Excessive Template Size (#{template_bytesize}) Max= #{cf_max_bytesize}")
```

We should raise Exceptions for failed validation instead of just printing text